modules/component-store/src/component-store.ts

```diff
@@ -32,6 +32,14 @@ export interface SelectConfig {
   debounce?: boolean;
 }
 
+export interface OnStoreInit {
+  readonly ngrxOnStoreInit: () => void;
+}
+
+export interface OnStateInit {
+  readonly ngrxOnStateInit: () => void;
```

Comment:
Just a question about this method, the rest looks fine to me.
Should this also contain the current `state`?
A use case that I can think of would be to invoke an effect with something that exists in the state.

Reply:
I thought about this also, and I'm not strongly opposed to it. We would have to change the interface to use a generic then though, right? Or can we infer it some other way

Reply:
Oh right, I didn't think of that.  
I don't think we can infer its type, which also takes the usefulness of this enhancement away.
If we can't infer it, perhaps we should just leave it as is. We can always add it later, when there's a strong need for it.